File: test/page_objects/check/signInPage.js
// Page class for SignIn page
'use strict'
/* global element by */
/* global browser browser */

class CheckSignIn {
  constructor () {
    this.pin = element(by.css('.pin-entry'))
    this.school_pin = element(by.css('#school-pin'))
    this.pupil_pin = element(by.css('#pupil-pin'))
    this.sign_in_button = element(by.css('.button-start'))
  }

  load () {
    browser.get('#/sign_in')
  }

  login (pinSchool, pinPupil) {
    this.school_pin.send_keys(pinSchool)
    this.pupil_pin.send_keys(pinPupil)
  }
}

module.exports = new CheckSignIn()

File: pupil-spa/test/features/support/page_objects/sign_in_page.rb
class SignInPage < SitePrism::Page
  set_url '/sign-in'

  element :heading, '.heading-xlarge', text: "Multiplication tables check"
  element :global_header, '#global-header .header-logo'
  element :footer_link, '.footer-wrapper .footer-meta .copyright a'
  section :phase_banner, PhaseBanner, '.js-content .phase-banner'

  element :welcome_message, '.lede', text: "Enter your school password and pupil PIN, then click ‘Sign in’"

  element :logo, '.organisation-logo'
  element :first_letter, '#last-name'
  element :day, '#dob-day'
  element :month, '#dob-month'
  element :pin, '.pin-entry'
  element :school_pin, '#schoolPin'
  element :pupil_pin, '#pupilPin'
  element :sign_in_button, '.button-start'
  element :auto_complete_off, "form[autocomplete='off']"

  def login(pin_school, pin_pupil=nil)
    school_pin.set pin_school
    pupil_pin.set pin_pupil
  end

end
